src/Data/Entity.php

```diff
@@ -3,52 +3,60 @@
 
 abstract class Entity implements \JsonSerializable
 {
-    protected $properties = [];
-
     protected $id;
+    protected $createdAt;
+    protected $updatedAt;
+    protected $deleted = false;
 
-    private $data;
-
-    private $callMethods = ['get', 'set'];
+    public function getId()
+    {
+        return $this->id;
+    }
 
-    public function __construct($data = [])
+    public function setId($id)
```

Comment:
Each entity class should have a single method that transforms the input to correct type. It could be `__construct`, though I find this bit ugly:

```php
$entity = clone $this;
$entity->__construct($data);
```

I think I'd rather see the interface:

```php
protected function apply(array $data)
{
    $data = array_intersect_key($data, get_object_vars($this));
    // optionally do additional transformations before this loop
    foreach ($data as $key => $value) {
        $this->$key = $value;
    }
}
```